python/perf-kernels/flash-attention-seqlen-padded.py

```diff
@@ -476,15 +479,28 @@ def forward(ctx, q, k, v, causal, sm_scale, split_kernel=False):
         waves_per_eu = 2 if Lq == 128 else 3
         num_warps = 8 if Lq == 128 else 4
         pre_load_v = False if Lq == 128 else True
-        
+        grid = (triton.cdiv(q.shape[2], BLOCK_M), q.shape[0] * q.shape[1], 1)
         stage = 3 if causal else 1
-        need_padding = True if seqlen % BLOCK_M else False
-        extra_tokens_n = seqlen % BLOCK_N
+
+        # Compute if we need padding and how much
+        seqlen_k = k.shape[-2]
+
+        if seqlen_k < BLOCK_N:
```

Comment:
Why is this one not included by the following elif?

Reply:
The assignments are different. `extra_tokens_n` is set differently. I could have an `if` statement inside the elif but I wanted to make it explicit that there are three different cases that the kernel handles in 3 different ways.

Reply:
Oh I see. I did not notice the difference.